doc/users/next_whats_new/selector_improvement.rst

```diff
@@ -0,0 +1,39 @@
+Selectors improvement: rotation, aspect ratio correction and add/remove state
+-----------------------------------------------------------------------------
+
+The `~matplotlib.widgets.RectangleSelector` and
+`~matplotlib.widgets.EllipseSelector` can now be rotated interactively.
+The rotation is enabled or disable by striking the *r* key
+(default value of 'rotate' in *state_modifier_keys*) or by calling
+*selector.add_state('rotate')*.
+
+The aspect ratio of the axes can now be taken into account when using the
+"square" state. This is enable by specifying *use_data_coordinates='True'* when
+the selector is initialized.
+
+In addition to changing selector state interactively using the modifier keys
+defined in *state_modifier_keys*, the selector state can now be changed
+programmatically using the *add_state* and *remove_state* method.
+
+
+.. code-block:: python
+
+    import matplotlib.pyplot as plt
+    from matplotlib.widgets import RectangleSelector
+    import numpy as np
+
+    values = np.arange(0, 100)
+
+    fig = plt.figure()
+    ax = fig.add_subplot()
+    ax.plot(values, values)
+
+    selector = RectangleSelector(ax, print, interactive=True,
+                                 drag_from_anywhere=True,
+                                 use_data_coordinates=True)
+    selector.add_state('rotate') # alternatively press 'r' key
+    # rotate the selector interactively
+
+    selector.remove_state('rotate') # alternatively press 'r' key
+
+    selector.add_state('square')
```

Comment:
I am wondering of some of that documentation and the example could be moved or copied into the docstring so it would continue to be readily available – this look like the most important part of the updated docs, and I am worried  it would get lost in the changelog over time.